src/store/login/login.reducer.ts

```diff
@@ -0,0 +1,245 @@
+import {
+  LoginActionTypes,
+  LoginAction,
+  WalletName,
+  AccountData,
+} from "src/store/login/login.actions";
+import { AsyncTask } from "src/utils/types";
+// domain
+import { HermezWallet } from "src/domain/hermez";
+import { AuthSignatures } from "src/domain/local-storage";
+import { getAuthSignatures } from "src/persistence/local-storage";
+
+export interface LoginState {
+  step: Step;
+  // ToDo: This state is never set. The actions do not exist.
+  // networkNameTask: AsyncTask<string, string>;
+  addAccountAuthTask: AsyncTask<null, string>;
+  accountAuthSignatures: AuthSignatures;
+}
+
+// ToDo: This should be removed when we migrate login.view.jsx module to TS
+export enum STEP_NAME {
+  WALLET_SELECTOR = "wallet-selector",
+  ACCOUNT_SELECTOR = "account-selector",
+  WALLET_LOADER = "wallet-loader",
+  CREATE_ACCOUNT_AUTH = "create-account-auth",
+}
+
+export type Step =
+  | {
+      type: "wallet-selector";
+      // ToDo: Why don't we move walletName to LoginState? It's in three steps and can be undefined...
```

Comment:
Sounds good to me

Reply:
I won't do it now though. It's better to do it after moving `views/login/login.view.jsx` to TS.

Reply:
Mmmm not sure why it can be `undefined`, because the `walletName` should exist only after clicking uour login method (metamask, or wallet connect). Maybe it doesn't make sense to have it in this step only, but maybe in the other ones, but without the possibility to be `undefined`.

Reply:
I've removed `walletName` from `wallet-selector` and `account-selector` login steps, and removed the `undefined` from the `wallet-loader` step.